gtda/time_series/target.py

```diff
@@ -74,12 +75,18 @@ class Labeller(BaseEstimator, TransformerResamplerMixin):
     """
 
     _hyperparameters = {
-        'func': [types.FunctionType, None],
-        '_percentiles': [list, [numbers.Number, (0., 100.)]],
-        'n_steps_future': [int, [1, np.inf]]}
+        'width': {'type': int, 'in': Interval(1, np.inf, closed='left')},
+        'func': {'type': FunctionType},
+        'func_params': {'type': (dict, type(None))},
+        'percentiles': {
+            'type': (list, type(None)),
+            'of': {'type': Real, 'in': Interval(0, 100, closed='both')}},
+        'n_steps_future': {
+            'type': int, 'in': Interval(1, np.inf, closed='left')}
+    }
 
     def __init__(self, width=10, func=np.std,
-                 func_params={}, percentiles=None, n_steps_future=1):
+                 func_params=None, percentiles=None, n_steps_future=1):
```

Comment:
Same comment as previously: why not pass a callable?

Reply:
Could you write what that callable would be here? I don't understand the suggestion in the top-level review comment. Also note that callables as default parameters *can* format badly in sphinx.

Reply:
So, `func` is a callable, right? Imagine that you pass `func=f` and `func_params ={'bla':1, 'bla2':123}`.
This is equivalent (?) to `func=partial(f, bla=1, bla2=123)` with `func_params=None`.
This is not specific to this pr though- general observation when reading the code.

Reply:
Ah I see what you meant! You are quite right that `partial` can be used instead and `func_params` eliminated altogether. The only argument for keeping `func_params` that I can think about/recall is that it is a little more friendly perhaps to a grid search and to `set_params`. If you wanted to keep the same `func` but change one of the parameters, without `func_params` you would have to repeat `partial(f, a=...)` as many times as different values for `a` you want to test, while with `func_params` you wouldn't need to repeat `f`.

Reply:
sure, i forgot about this.You are completely right! My bad